src/os/xt/peer.js

```diff
@@ -481,6 +484,26 @@ os.xt.Peer.prototype.init = function() {
   this.pingTimer_.start();
 
   goog.log.fine(os.xt.Peer.LOGGER_, 'Initialized peer ' + this.group_ + '.' + this.id_ + ' "' + this.title_ + '"');
+  this.processInitialMessages();
+};
+
+
+/**
+ * Processes messages already available on startup
+ * @protected
+ */
+os.xt.Peer.prototype.processInitialMessages = function() {
+  var priv = ['xt', this.group_, this.id_, ''].join('.');
+  var pub = ['xt', this.group_, 'public', ''].join('.');
+
+  var storage = this.storage_;
+  var i = storage.length;
+  while (i--) {
+    var key = storage.key(i);
+    if (key && (key.startsWith(pub) || (key.startsWith(priv) && os.xt.Peer.notThese_.indexOf(key) === -1))) {
```

Comment:
The `os.xt.Peer.notThese_.indexOf(key) === -1` check here should be looking for the last part of the key in that array, not the entire key. Currently that condition will always be true and extra messages will be processed.